app/models/target.rb

```diff
@@ -18,9 +18,16 @@
 #  index_targets_on_user_id   (user_id)
 #
 class Target < ApplicationRecord
+  belongs_to :user
+  belongs_to :topic
+  validate :targets_count, on: :create, unless: -> { user.nil? }
   validates :title, presence: true
   validates :radius, presence: true, numericality: { greater_than: 0 }
   validates :lat, :lon, presence: true, numericality: true
-  belongs_to :user
-  belongs_to :topic
+
+  def targets_count
+    return unless user.targets.count >= 3
+
+    errors.add(:base, "You can't create more than 3 targets")
```

Comment:
just a suggestion, maybe this error can be related to the user field
```suggestion
    errors.add(:user, "You can't create more than 3 targets")
```